### PR TITLE
Add `pcscd` install instructions for hardware key support

### LINKED ISSUE #32110: [Docs] pcscd is required to enable PIV
Docs page: https://goteleport.com/docs/access-controls/guides/hardware-key-support/

PIV support seems to require `pcscd` daemon installed on Ubuntu. If the package is not installed `tsh` seems to fail with 

```
ERROR: failed to get a YubiKey private key
connecting to pscs: the Smart card resource manager is not running
```

After the package installation `apt install pcscd` the issue seems to be resolved. Our docs don't mention it anywhere.

### PR DESCRIPTION
**Pull request by Joerger**
Closes https://github.com/gravitational/teleport/issues/32110

### CODE REVIEW

File: docs/pages/access-controls/guides/hardware-key-support.mdx [hunk 1]
```diff
@@ -77,6 +77,9 @@ like `tctl edit`. With touch required, hardware key support provides better secu
   so we recommend only using one PIV application at a time.
 </Admonition>
 
+- Running Smart Card daemon:
+  - Ships with MacOS and Windows, no install necessary.
+  - On Linux: `apt get pcscd`. Check status with `systemctl status pcscd`.
```
Comment (jakule) [inline docs/pages/access-controls/guides/hardware-key-support.mdx:82]: `apt` doesn't work on all Linux distributions, only on Debian-based as Debian and Ubuntu. I'd mention that this is an Ubuntu-only command. I'd also add something like "consult your distribution manual on how to install pcscd" for those who are running something different.

File: docs/pages/access-controls/guides/hardware-key-support.mdx [hunk 2]
```diff
@@ -77,6 +77,9 @@ like `tctl edit`. With touch required, hardware key support provides better secu
   so we recommend only using one PIV application at a time.
 </Admonition>
 
+- Running Smart Card daemon:
```
Comment (ptgott) [inline docs/pages/access-controls/guides/hardware-key-support.mdx:80]: This is the only mention of "Smart Card daemon" on this page, so I would add a sentence about how hardware key support uses the smart card daemon. (Is "Smart Card daemon" the product name, or would it make sense to lowercase it?)

File: docs/pages/access-controls/guides/hardware-key-support.mdx [hunk 3]
```diff
@@ -77,6 +77,9 @@ like `tctl edit`. With touch required, hardware key support provides better secu
...
+  - Ships with MacOS and Windows.
+  - On Linux, consult your distribution manual on how to install `pcscd`.
```
Comment (ptgott) [inline docs/pages/access-controls/guides/hardware-key-support.mdx:82]: I would make it explicit that `pcscd` is the smart card daemon

**Review by jakule — APPROVED** [review]

File: docs/pages/access-controls/guides/hardware-key-support.mdx (hunk 2 above)
Comment (Joerger) [reply docs/pages/access-controls/guides/hardware-key-support.mdx]: Changed to "smart card driver" which is more universal

**Review by ptgott — APPROVED** [review]

**Comment by ptgott** [comment]
@Joerger Is this a v15 behavior change, or should we add backport labels?

**Pull request merged**